Doc/library/socket.rst

```diff
@@ -896,13 +896,13 @@ The :mod:`socket` module also offers various network-related services:
 
 .. function:: getfqdn([name])
 
-   Return a fully qualified domain name for *name*. If *name* is omitted or empty,
-   it is interpreted as the local host.  To find the fully qualified name, the
+   Return a fully qualified domain name for *name*. If *name* is empty or equal to
+   ``'0.0.0.0'``, the hostname from :func:`gethostname` is returned.
```

Comment:
Is it so? Looking at the sources, it tries to return one of names returned by `gethostbyaddr(gethostname())`.

The new wording also leaves unclear what `getfqdn()` returns.

Reply:
The source/behavior was changed in https://github.com/python/cpython/pull/100375 after this PR was opened, so wording definitely does need updating.

Regarding the unclear new wording, could you expand on that? My interpretation is that previously the entire paragraph needs to be read through before the return of `getfqdn()`, but now it's right at the top: `If *name* is empty ...`

Reply:
Now it starts from the rare corner case (why would anyone pass an empty name to `getfqdn()`?), but the documentation for `getfqdn()` without arguments is removed.

Reply:
You are very correct, leaving out the empty argument is my mistake. What do you think about this instead?

```
Return a fully qualified domain name for *name*.
To find the fully qualified name, ...
...
... it is returned unchanged.
If *name* is empty, or equal to "0.0.0.0," "::", or "",
the hostname from :func:`gethostname` is returned.
```

There's also an option of changing `getfqdn([name])` to `getfqdn(name='')`, but not sure if it helps with improving the wording much.

Reply:
I think that both changing the wording and changing the signature are worth to do.

Reply:
Done, let me know if there's anything that looks off. For reference, [here's](https://cpython-previews--93451.org.readthedocs.build/en/93451/library/socket.html#socket.getfqdn) the preview of `getfqdn`.